src/Compilers/VisualBasic/Portable/Binding/Binder.vb

```diff
@@ -561,14 +555,33 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return specialMemberSymbol
         End Function
 
-        Friend Shared Function GetSpecialTypeMember(assembly As AssemblySymbol, member As SpecialMember, <Out> ByRef useSiteInfo As UseSiteInfo(Of AssemblySymbol)) As Symbol
+        Friend Shared Function GetSpecialTypeMember(assembly As AssemblySymbol, member As SpecialMember, <Out> ByRef useSiteInfo As UseSiteInfo(Of AssemblySymbol), Optional isOptional As Boolean = False) As Symbol
             Dim specialMemberSymbol As Symbol = assembly.GetSpecialTypeMember(member)
 
-            If specialMemberSymbol Is Nothing Then
+            If specialMemberSymbol IsNot Nothing Then
+                useSiteInfo = GetUseSiteInfoForMemberAndContainingType(specialMemberSymbol)
+
+                If useSiteInfo.DiagnosticInfo IsNot Nothing Then
+                    ' Report errors only for non-optional members
+                    If isOptional Then
+                        Dim severity = useSiteInfo.DiagnosticInfo.Severity
+
+                        ' If the member is optional and bad for whatever reason ignore it
+                        If severity = DiagnosticSeverity.Error Then
```

Comment:
>If severity = DiagnosticSeverity.Error Then [](http://example.com/codeflow?start=24&length=43)

I think this check isn't necessary in VB because  `useSiteInfo.DiagnosticInfo` can be only an error in VB.